### PR TITLE
Update demo upload colab to use `--one_shot`

### PR DESCRIPTION
**Pull request by bileschi**
`--one_shot` is the preferred workflow in this case, where the training is already complete before the upload begins.
`--one_shot` upload was introduced in tensorboard 2.3.0

Tested: ran through the notebook on Jupyter on my laptop using `python 3.7.3` and `tensorflow 2.3.0`

### CODE REVIEW

**Comment by bileschi** [comment]
I updated this notebook by opening it in Jupyter, editing there, and saving (after stripping results).  I see that it has changed much of the formatting, making the diff very difficult to read, even with the ReviewNB tool.  Is there a different flow I should have used here?

**Comment by bileschi** [comment]
Ok, the answer is that I should have used nbfmt to format the output.  This is the recipe I followed, based on what it looks like the github action was doing.
```
$ nbfmt_version="78692e5013fe070a33c7501245157b395c2e7e84"
$ wget "https://raw.githubusercontent.com/tensorflow/docs/${nbfmt_version}/tools/nbfmt.py"
$ chmod +x ./nbfmt.py
$ ./nbfmt.py --ignore_warn tensorboard/docs/tbdev_getting_started.ipynb ;
```

File: docs/tbdev_getting_started.ipynb [hunk 1]
```diff
@@ -1,235 +1,272 @@
 {
```
Comment (GalOshri) [inline docs/tbdev_getting_started.ipynb]: <p>This notebook is primarily run in Colab (users reach it from "Example Colab" on https://tensorboard.dev). I don't think we should significantly increase the length of the tutorial with this section.</p><p> </p><p>What if we changed this to a one line statement in the next section: "When running in a Jupyter notebook, use <code>tensorboard dev list</code> to authenticate".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (GalOshri) [inline docs/tbdev_getting_started.ipynb]: <p>The newline formatting disappeared from this section when viewed in Colab.</p><p> </p><p>I suggest rewording the new part to: "The <code>--one_shot</code>  flag causes the uploader to exit when it has uploaded the entire <code>logdir</code> . This should be used when the experiment has already completed. Without this flag, the uploader will keep running until it is stopped, in order to read new data from the directory during ongoing training."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (GalOshri) [inline docs/tbdev_getting_started.ipynb]: <p>This doesn't currently work in Colab as I don't think TF/TB 2.3 is the native. I am not sure if this happens automatically or whether we need to do something. A workaround is to upgrade the version of TensorBoard at the start of the notebook, but it would be better if we can avoid it.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

File: docs/tbdev_getting_started.ipynb [hunk 2]
```diff
@@ -12,7 +12,7 @@
     },
```
Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>instead of "provides a managed TB experience" can we rephrase as:</p><p> </p><p>TensorBoard.dev is a free, public TensorBoard service that enables you upload and share your ML experiment results with everyone.</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>"TensorBoard"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>s/colab/Colab</p><p> </p><p>I don't think you need the "stdin" explanation. Just rephrase the second paragraph as</p><p> </p><p>"This step requires you to auth in your shell console, outside of Jupyter. In your console, execute the following command:"</p><p> </p><p>&lt;command&gt;</p><p> </p><p>As part of this flow, you will be provided with an authorization code. This code is required to consent to the Terms of Service. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>Uploading the TensorBoard logs will give<strong> you a URL </strong>that can be shared with anyone.</p><p> </p><p>Uploaded TensorBoards are public, so do not upload sensitive data.</p><p> </p><p>&lt;Also ...&gt;</p><p> </p><p>This sentence is a run on:</p><p>    The uploader will exit when it has completed uploading the entire <code>logdir</code>, this is the function of the <code>--one_shot</code> flag. </p><p> </p><p>Rephrase to:</p><p> </p><p>The uploader will exit when the entire logdir has been uploaded. (This is what the --one_shot flag specifies.)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>Uploading the TensorBoard logs will give<strong> you a URL </strong>that can be shared with anyone.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>Remove "Note that".</p><p> </p><p>Rephrase final sentence to:</p><p> </p><p>"You can list all the experiments you have uploaded by using</p><p> </p><p><code>tensorboard dev list</code></p><p> "</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

**Comment by bileschi** [comment]
This PR is blocked on public colab support for TensorFlow >= 2.3.0
@Googlers, this is bug b/162521811

File: docs/tbdev_getting_started.ipynb (hunk 2 above)
Comment (bileschi) [reply docs/tbdev_getting_started.ipynb:12]: done

Comment (bileschi) [reply docs/tbdev_getting_started.ipynb:12]: done

Comment (bileschi) [reply docs/tbdev_getting_started.ipynb:12]: done

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>"features which are only available for versions &gt;= 2.3.0."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>Is something funky about the spacing in the construction of the TensorBoard callback? I don't see indents.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>s/commandn/command</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>specifies.)</p><p> </p><p>(Period at end.)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>s/baord/board</p><p> </p><p>Also ensure there's no newline before the command?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (bileschi) [reply docs/tbdev_getting_started.ipynb:12]: <p>done</p>

Comment (bileschi) [reply docs/tbdev_getting_started.ipynb:12]: <p>That line was over 80 char.  I've broken it into two lines.</p>

Comment (bileschi) [reply docs/tbdev_getting_started.ipynb:12]: <p>done</p>

Comment (bileschi) [reply docs/tbdev_getting_started.ipynb:12]: <p>Done.</p>

Comment (bileschi) [reply docs/tbdev_getting_started.ipynb:12]: <p>Done.</p>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>specifies.)</p><p> </p><p>(Period goes before the end paren.)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>It seems a clearer explanation is:</p><p> </p><p>"Each individual upload has a unique experiment ID. This means that if you start a new upload with the same directory, you will get a new experiment ID."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>At the end of this, consider adding a screenshot of TensorBoard.dev as seen in the browser.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (bileschi) [reply docs/tbdev_getting_started.ipynb:12]: done.

Comment (bileschi) [reply docs/tbdev_getting_started.ipynb:12]: <p>Done.</p>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>Extra newline before "You can list...".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (manivaradarajan) [inline docs/tbdev_getting_started.ipynb:12]: <p>Is there an extra newline here before "In the above"?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tensorboard/pull/3920/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (bileschi) [reply docs/tbdev_getting_started.ipynb:12]: This will not render in the notebook
image

Comment (bileschi) [reply docs/tbdev_getting_started.ipynb:12]: This will only appear in the pre-markdown format as a line break.  It does not appear as rendered for the user.

image
